### PR TITLE
Upgrade lodash to v4

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

We're currently using lodash@v2, which is outdated, but generally works.

However, lodash@v2 is not exported as Node modules, hence we are using the (now deprecated) Node-compatibility version [`lodash-node`](https://www.npmjs.com/package/lodash-node). This means we have awkward module mappings in our [Webpack](https://github.com/guardian/frontend/pull/20475/commits/d73302e46f7fb55d797ac0f0de36eb633841cf7d#diff-11e9f7f953edc64ba14b0cc350ae7b9dL68) and [Jest](https://github.com/guardian/frontend/pull/20475/commits/d73302e46f7fb55d797ac0f0de36eb633841cf7d#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L148) configurations.

Ultimately this manifests itself as two issues:

- We can't upgrade Jest beyond v23.1, due to a dependency on lodash interfering with our module mapping
- we have had to exclude tests for our custom ESLint plugins, which fall over due to the Jest module mapping

This change upgrades lodash to v4, so we can use the proper lodash library and no longer need `lodash-node` or the module mappings.

## What is the value of this and can you measure success?

More up to date modules, fewer awkward workarounds that are causing problems. Also lodash@v4 has better Flow types.

Most of the changes are paths, the exceptions being:

- `flatten` is no longer deep by default, and no longer takes a callback (see [my comment below](https://github.com/guardian/frontend/pull/20475#discussion_r222975876)). Deep flattens are replaced by `flattenDeep`
- `template` no longer receives a `data` object as the second argument, but returns a `compiled` function to which we must pass the `data` object
- `pick` no longer takes a callback. References with callbacks are replaced by `pickBy`
- `uniq` no longer takes a callback. References with callbacks are replaced by `uniqBy`

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const getSizeOpts = (sizesByBreakpoint: Object): Object => {
     const sizeMapping = buildSizeMapping(sizesByBreakpoint);
     // as we're using sizeMapping, pull out all the ad sizes, as an array of arrays
     const sizes = uniq(
-        flatten(sizeMapping, true, map => map[1]),
+        flatten(sizeMapping.map(size => size[1])),
```
Comment (SiAdcock) [inline static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: For sanity, does this look okay to everyone? Especially interested in the opinions of @guardian/commercial-dev 😄

**Comment by SiAdcock** [comment]
Since Snyk is flagging a vulnerability with lodash@v3, I'm going to try upgrading straight to lodash@v4 instead. Please hang tight!

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js (hunk 1 above)
Comment (SiAdcock) [reply static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: Good shout. Ads aren't appearing for me locally on this branch or on master. I'll test it on CODE next week. The input / output looks the same locally vs in production

Comment (SiAdcock) [reply static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: Looks good in CODE

**Review by AWare — APPROVED** [review]
rollercoaster gif from giphy
(when flow is happy)

File: static/src/javascripts/projects/common/modules/email/email.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { trackNonClickInteraction } from 'common/modules/analytics/google';
 import { inlineSvg } from 'common/views/svgs';
 import { getUserFromApi, isUserLoggedIn } from 'common/modules/identity/api';
 import userPrefs from 'common/modules/user-prefs';
-import uniq from 'lodash/arrays/uniq';
+import uniq from 'lodash/uniq';
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/email/email.js]: Should this be converted to `uniqBy`?

Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/email/email.js]: Oh forget that, I see it takes a collection not a callback so `uniq` is 👌

**Review by GHaberis — APPROVED** [review]
👍

**Pull request merged**